lib/upgrade/migrator/migrate.go

```diff
@@ -891,6 +890,32 @@ func (m *migrationHelper) storeV3Resources(data *MigrationData) error {
 	return nil
 }
 
+func toStorage(r converters.Resource) converters.Resource {
+	// Set timestamp and UID.
+	r.GetObjectMeta().SetCreationTimestamp(metav1.Now())
+	r.GetObjectMeta().SetUID(uuid.NewUUID())
+
+	// Some types require additional processing when converting to storage.
+	switch r.(type) {
+	case *apiv3.GlobalNetworkPolicy:
+		// For GNP, we need to adjust the name before storage.
+		r.GetObjectMeta().SetName(convertPolicyNameForStorage(r.GetObjectMeta().GetName()))
+	case *apiv3.NetworkPolicy:
+		// For NP, we need to adjust the name before storage.
+		r.GetObjectMeta().SetName(convertPolicyNameForStorage(r.GetObjectMeta().GetName()))
+	}
+
+	return r
+}
+
+func convertPolicyNameForStorage(name string) string {
```

Comment:
can we move this to https://github.com/projectcalico/libcalico-go/blob/master/lib/upgrade/converters/names.go#L63 with other name conversions? converters package is already imported here, so won't even need a new import

Reply:
@gunjan5 suppose we could - it is duplicating code, but I was trying not to make this a public function because I don't think it should be made public.

I'd rather leave as is.